runtime/gc_include/ObjectAccessBarrierAPI.hpp

```diff
@@ -411,7 +411,8 @@ class MM_ObjectAccessBarrierAPI
 #if defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER)
 		return vmThread->javaVM->memoryManagerFunctions->j9gc_objaccess_structuralCompareFlattenedObjects(vmThread, valueClass, lhsObject, rhsObject, startOffset);
 #else /* defined(J9VM_GC_ALWAYS_CALL_OBJECT_ACCESS_BARRIER) */
-		if (j9gc_modron_readbar_none != _readBarrierType) {
+		bool hasReferences = J9CLASS_HAS_REFERENCES(valueClass);
+		if (hasReferences && (j9gc_modron_readbar_none != _readBarrierType)) {
```

Comment:
expand this with '|| (j9gc_modron_wrtbar_always == _writeBarrierType))' similarly to the other site

Reply:
even though this is a readonly operation?

Reply:
ah, good point. then, no.
the other one was a mix of both reads and writes, so I was hasty to assume this was the same case.